framework/src/preconditioners/FieldSplitPreconditioner.C

```diff
@@ -72,7 +72,7 @@ FieldSplitPreconditioner::FieldSplitPreconditioner(const InputParameters & param
       for (unsigned int j = 0; j < n_vars; j++)
         (*cm)(i,j) = 1; // full coupling
   }
-  _fe_problem.setCouplingMatrix(cm);
+  _fe_problem.setCouplingMatrix(std::move(cm));
```

Comment:
Related to my other comment, maybe `setCouplingMatrix` should be taking its argument by rvalue-reference `(&&)` to force people to `std::move` into it.